src/webapp/pages/form-page/disease-outbreak-event/utils/mapFormStateToEntityData.ts

```diff
@@ -178,20 +179,9 @@ function getValidDateValuesByFieldIdFromFields(
         establishCoordination: getFromAllFields(diseaseOutbreakEventFieldIds.establishCoordination),
     };
 
-    return Object.keys(dateValues).reduce(
-        (acc: Record<DateFieldIdsToValidate, Date>, key: string) => {
-            const dateFieldId = key as DateFieldIdsToValidate;
-
-            const dateValue = getDateFieldValue(dateFieldId, allFields);
-            if (dateValue === null) {
-                throw new Error(`Invalid date value for field: ${dateFieldId}`);
-            } else {
-                return {
-                    ...acc,
-                    [dateFieldId]: dateValue,
-                };
-            }
-        },
-        {} as Record<DateFieldIdsToValidate, Date>
-    );
+    if (Object.values(dateValues).every((dateValue): dateValue is Date => dateValue !== null)) {
```

Comment:
Proposal: Add the check for null in getFromAllFields. And then you can remove this final check altogether.